Packs/ExtraHop/Integrations/ExtrahopRevealXEventCollector/ExtrahopRevealXEventCollector.py

```diff
@@ -0,0 +1,471 @@
+                       "sort": [{"direction": "asc", "field": "mod_time"}]}
+
+    events, next_run = fetch_extrahop_detections(client, advanced_filter, last_run, max_events)
+    demisto.info(f"Extrahop next_run is {next_run}")
+    return events, next_run
+
+
+def get_events(client: Client, args: dict, max_events: int) -> tuple[list, CommandResults]:
+    """
+    Inner Test Function to make sure the integration works
+    Args:
+        client: ExtraHop client to be used.
+        args: command arguments.
+        max_events: Max events to fetch.
+
+    Returns: Tuple that contains events that been fetched and the Command results.
+    """
+    first_fetch = arg_to_datetime(args.get("first_fetch")) or get_current_time()
+
+    # if the user limits in the get events arguments
+    last_run = {
+        'detection_start_time': int(first_fetch.timestamp() * 1000),
+        'offset': 0,
+    }
+
+    output, _  = fetch_events(client, last_run, max_events)
+    human_readable = prepare_list_detections_output(output)
+
+    command_results = CommandResults(
+        readable_output=human_readable,
+        outputs=output,
+        outputs_prefix="ExtraHop.Detections",
+    )
+    return output, command_results
+
+
+def main():
+    """main function, parses params and runs command functions"""
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+
+    demisto.debug(f"Command being called is {command}")
+    try:
+        base_url = params.get("server_url")
+        verify_certificate = not argToBoolean(params.get("insecure", False))
+        client_id = params.get("credentials", {}).get("identifier")
+        client_secret = params.get("credentials", {}).get("password")
+        use_proxy: bool = params.get('proxy', False)
+        max_events = arg_to_number(params.get('max_events_per_fetch')) or MAX_FETCH_LIMIT
+
+        client = Client(base_url=base_url,
+                        verify=verify_certificate,
+                        client_id=client_id,
+                        client_secret=client_secret,
+                        use_proxy=use_proxy,
+                        ok_codes=(200, 201, 204))
+
+        if command == "test-module":
+            # Command made to test the integration
+            result = test_module(client)
+            return_results(result)
+        elif command == "fetch-events":
+            last_run = demisto.getLastRun()
+            events, next_run = fetch_events(client, last_run, max_events)
+            if len(events):
+                demisto.debug(f'Sending {len(events)} events.')
+                send_events_to_xsiam(events=events, vendor=VENDOR, product=PRODUCT)
+            demisto.setLastRun(next_run)
+            demisto.debug(f'Successfully saved last_run= {demisto.getLastRun()}')
+        elif command == "revealx-get-events":
+            max_events = arg_to_number(args.get("max_events")) or max_events
```

Comment:
You can change this arg to be limit
Also you're passing the args, so it will make more sense to move this arg to the command itself, instead of the main